### PR TITLE
breaking change: remove fixture-generator

### PR DESCRIPTION
**Pull request by craigbilner**
This PR removes `fixture-generator` and cleans up the mash-up of author-profile and topic fixtures. It also makes stories/tests use the expected data structures in prod rather than handmade which results in stale fixtures.

`MockFixture`: allows a promise based API to be used with storybook in order to show the story when the mocks are ready
`make-mocks`: is the use of Apollo to auto-generate data based on the `gql` queries
`generate-queries`: is a simple helper to generate pages of data
`makeParams`: there are currently two helpers in `provider-test-tools` for `author-profile` and `topic`. They wrap around `make-mocks` to provide sensible defaults for values and types but allow overriding for different scenarios. In addition they provide duplicate queries for errors/retries and placeholder images for the stubbed data. When creating a story/test you can curate the data you need by carefully deciding whether to mock a value or the type itself. The added benefit is extra safety by making stories/tests intrinsically linked to the server while still loosely coupled in their implementation.

n.b. if you don't provide stub data it will result in Apollo's own stub, e.g. a `string` would become `Hello World`.

next: Article needs to be moved over so we don't have the current problem where queries aren't lining up with the fixtures

* refactor: all queries now have `__typename` added to them
* refactor: hoisted lead assets to short circuit provider selection
* feature: author profile and topic showcases now use new pattern of create mocks with the `gql` query itself using the `MockFixture` helper
* chore: alphabetised the queries (added benefit is that creating Apollo mocks which need exact query/variable matches isn't as surprising)
* chore: remove `fixture-generator`

### CODE REVIEW

File: packages/author-profile/author-profile.showcase.js [hunk 1]
```diff
@@ -18,148 +23,249 @@ const preventDefaultedAction = decorateAction =>
...
-  isLoading: false,
   onArticlePress: preventDefaultedAction(decorateAction)("onArticlePress"),
   onTwitterLinkPress: preventDefaultedAction(decorateAction)(
     "onTwitterLinkPress"
   ),
-  page: 1,
-  pageSize,
-  refetch: preventDefaultedAction(decorateAction)("refetch"),
-  slug
+  refetch: preventDefaultedAction(decorateAction)("refetch")
```
Comment (colincclark) [inline packages/author-profile/author-profile.showcase.js]: the refetch events have lost their ability to refresh in the stories

File: packages/provider-test-tools/__tests__/fixture-gen.test.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { iterator } from "@times-components/test-utils";
+import mm from "../src/make-mocks";
```
Comment (colincclark) [inline packages/provider-test-tools/__tests__/fixture-gen.test.js]: 😋

File: packages/article-list/article-list.showcase.js [hunk 3]
```diff
@@ -20,62 +31,201 @@ const preventDefaultedAction = decorateAction =>
...
+const page = 1;
+const pageSize = 10;
+const slug = "deborah-haynes";
+
 export default {
   name: "Composed/Article List",
   children: [
     {
       type: "story",
       name: "Default with images",
```
[the remark this answers is not in the document]
Comment (craigbilner) [reply packages/article-list/article-list.showcase.js]: Potentially we can use a random image API

File: packages/article-list/article-list.showcase.js [hunk 4]
```diff
@@ -20,62 +31,201 @@ const preventDefaultedAction = decorateAction =>
...
+              first: pageSize,
+              imageRatio: articleImageRatio,
+              skip: (iteration - 1) * pageSize,
+              slug
+            }),
+            count,
+            pageSize,
+            slug
+          })}
+          render={mocks => (
```
[the remark this answers is not in the document]
Comment (craigbilner) [reply packages/article-list/article-list.showcase.js]: happy to change it, what would be preferable?

File: packages/article-list/article-list.showcase.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (craigbilner) [reply packages/article-list/article-list.showcase.js]: updated :+1:

**Comment by aronshamash** [comment]
Nice work. In general I think this is good. We can alway iterate over it.

**Pull request merged**